src/common/storage/backends/s3/Client.js

```diff
@@ -110,20 +107,44 @@ define([
         return dataInfo;
     };
 
+    S3Storage.prototype.putStream = async function(filename, stream) {
+        await this.checkStreamsInBrowser();
+        const s3Client = await this.getS3Client();
+        const params = await this.getUploadParams(s3Client, filename, stream);
+        try {
+            await s3Client.upload(params).promise();
+        } catch (err) {
+            throw new Error(`Unable to upload ${stream}: ${err.message}`);
+        }
+        const dataInfo = await this.stat(filename);
+        this.logger.debug(`Successfully uploaded file ${filename} to the S3 server using stream`);
+        return dataInfo;
+    };
+
+    S3Storage.prototype.getUploadParams = async function(s3Client, filename, body) {
+        await this.createBucketIfNeeded(s3Client);
+        this.logger.debug(`Created bucket ${this.bucketName}`);
+        return {
+            Body: require.isBrowser ? new Blob([body]) : body,
+            Bucket: this.bucketName,
+            Key: filename,
+        };
+    };
+
     S3Storage.prototype.deleteDir = async function (dirname) {
         const s3Client = await this.getS3Client();
         const {Contents} = await s3Client.listObjectsV2({
             Bucket: this.bucketName,
             MaxKeys: 1000,
             Prefix: dirname
-        });
+        }).promise();
```

Comment:
Switching to `.promise` rather than the `promisifyMethod` is a nice refactor. It would be a little cleaner to move this type of refactor to its own PR.